v4 Beta: "Module not found: Default condition should be last one"

### Checklist

- [X] I have looked into the [Readme](https://github.com/auth0/node-auth0#readme), [Examples](https://github.com/auth0/node-auth0/blob/master/EXAMPLES.md), and [FAQ](https://github.com/auth0/node-auth0/blob/master/FAQ.md) and have not found a suitable solution or answer.
- [X] I have looked into the [API documentation](https://auth0.github.io/node-auth0/) and have not found a suitable solution or answer.
- [X] I have searched the [issues](https://github.com/auth0/node-auth0/issues) and have not found a suitable solution or answer.
- [X] I have searched the [Auth0 Community](https://community.auth0.com) forums and have not found a suitable solution or answer.
- [X] I agree to the terms within the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md).

### Description

Attempting to import node-auth0 v4.0.0-beta.7 into a Next.js app fails with the error "Module not found: Default condition should be last one".
This appears to be a constraint enforced by webpack:
https://github.com/webpack/enhanced-resolve/blob/3a28f47788de794d9da4d1702a3a583d8422cd48/lib/util/entrypoints.js#L472-L476

And reading Node.js documentation indicates that it's for a valid reason, in that the order of keys matters and an implementation must choose the first matching entry (and default always matches):
https://nodejs.org/api/packages.html#conditional-exports

The change that I believe is necessary is this:
https://github.com/auth0/node-auth0/compare/beta...cysp:node-auth0:beta-package-json-import-order

### Reproduction

1. add node-auth0 v4.0.0-beta.7 to next.js app (or probably any project that uses webpack)
2. build app

### Additional context

_No response_

### node-auth0 version

4.0.0-beta.7

### Node.js version

18.17.1

--- a/package.json
+++ b/package.json
@@ -8,12 +8,12 @@
   "exports": {
     ".": {
       "import": {
-        "default": "./dist/esm/index.js",
-        "types": "./dist/esm/index.d.ts"
+        "types": "./dist/esm/index.d.ts",
+        "default": "./dist/esm/index.js"
       },
       "require": {
-        "default": "./dist/cjs/index.js",
-        "types": "./dist/cjs/index.d.ts"
+        "types": "./dist/cjs/index.d.ts",
+        "default": "./dist/cjs/index.js"
       }
     }
   },

